content/en/docs/contributing/release-process.md

```diff
@@ -27,181 +27,211 @@ following conditions:
+page if a step is missing or if it is outdated.
 {{% /pageinfo %}}
 
-The process for cutting a minor release is as follows:
-
-1. Ensure upgrading document exists. See for example, see
-   [upgrading-1.0-1.1](https://cert-manager.io/docs/installation/upgrading/upgrading-1.0-1.1/)
-   (not necessary for alpha and patch releases)
-
-2. Create or update the release branch
-
-    If this is the first alpha release (`alpha.0`), then you will need to create
-    the release branch:
+1. Make sure to note which type of release you are doing. That will be helpful
+   in the next steps.
 
-    ```bash
-    # Must be run from the cert-manager repo folder.
-    git fetch --all
-    git checkout -b release-1.0 origin/master
-    ```
+    | Type of release          | Example of git tag |
+    | ------------------------ | ------------------ |
+    | initial alpha release    | `v1.3.0-alpha.0`   |
+    | subsequent alpha release | `v1.3.0-alpha.1`   |
+    | beta release             | `v1.3.0-beta.0`    |
+    | final release            | `v1.3.0`           |
+    | patch release            | `v1.3.1`           |
+    </br>
 
-    If there has already been an alpha release, the release branch will
-    already exist, so you will need to update it with the latest commits
-    from the master branch, as follows:
+1. **(final release only)** Make sure that a PR with the new upgrade document
+   is ready to be merged on
+   [cert-manager/website](https://github.com/cert-manager/website). See for
+   example, see
+   [upgrading-1.0-1.1](https://cert-manager.io/docs/installation/upgrading/upgrading-1.0-1.1/).
 
-    ```bash
-    # Must be run from the cert-manager repo folder.
-    git fetch --all
-    git branch --force release-1.0 origin/release-1.0
-    git checkout release-1.0
-    git merge --ff-only origin/master
-    ```
+2. Create or update the release branch:
 
-3. Push it to the `jetstack/cert-manager` repository
+    - **(initial alpha only)** Create the release branch:
 
-     **Note 1**: run `git remote -v` to check that `origin` points to the
-     upstream <https://github.com/jetstack/cert-manager.git>.
+       ```bash
+       # Must be run from the cert-manager repo folder.
+       git fetch --all
+       git checkout -b release-1.0 origin/master
+       ```
 
-     **Note 2:** if the branch doesn't already exist, you will need to have the
-     `write` role on the GitHub project to be able to push to the release
-     branch.
+    - **(subsequent alpha, beta, final, and patch releases only)**
+      You need to update it with the latest commits from the master branch, as
+      follows:
 
-     ```bash
-     # Must be run from the cert-manager repo folder.
-     git push --set-upstream origin release-1.0
-     ```
+       ```bash
+       # Must be run from the cert-manager repo folder.
+       git fetch --all
+       git branch --force release-1.0 origin/release-1.0
+       git checkout release-1.0
+       git merge --ff-only origin/master
```

Comment:
This is only for new minor releases, not for patches where the cherry-picking is done through comments in the GitHub UI right?

Reply:
You are totally right, I added a note about this point in ebb0e6f.